backend.py

```diff
@@ -1707,6 +1713,8 @@ def verifyRepos(sources, ui):
         if not repo_good:
             raise RuntimeError("Unable to access repository (%s, %s)" % (i['media'], i['address']))
 
+    mounter.umount()
```

Comment:
I think all of these mount/umount calls should be in a `try...finally` (or use a context manager like I suggested elsewhere).

Reply:
I'm not really familiar with context manager but I'll look at it (I've done already... only not so often so I don't fully remember, no problems). But not sure how context manager interacts with exceptions... oh... the leave part is called like a `finally` part... got it.